corehq/apps/reports/standard/deployments.py

```diff
@@ -162,41 +172,43 @@ def sort_filter(self):
 
     def get_sorting_block(self):
         sort_prop_name = 'prop_name' if self.selected_app_id else 'alt_prop_name'
-        res = []
-        #the NUMBER of cols sorting
-        sort_cols = int(self.request.GET.get('iSortingCols', 0))
-        if sort_cols > 0:
-            for x in range(sort_cols):
-                col_key = 'iSortCol_%d' % x
-                sort_dir = self.request.GET['sSortDir_%d' % x]
-                col_id = int(self.request.GET[col_key])
-                col = self.headers.header[col_id]
-                sort_prop = getattr(col, sort_prop_name) or col.prop_name
-                if x == 0:
-                    self.primary_sort_prop = sort_prop
-                if self.selected_app_id:
-                    sort_dict = {
-                        sort_prop: {
-                            "order": sort_dir,
-                            "nested_filter": {
-                                "term": {
-                                    self.sort_filter: self.selected_app_id
-                                }
-                            }
-                        }
+        block = []
+        for col in self.datatables_params.order:
+            col_ind = col['column']
+            sort_dir = col['dir']
+            dt_column_obj = self.headers.header[col_ind]
+            sort_prop = getattr(dt_column_obj, sort_prop_name) or dt_column_obj.prop_name
+            if col_ind == 0:
+                # this feels like a bit of a hack, but kept it in from the original sorting block
+                # prior to bootstrap 5 migration. could use a second look in the future
+                self.primary_sort_prop = sort_prop
+            if self.selected_app_id:
+                sort_dict = self._get_selected_app_sort_dict(sort_prop, sort_dir)
+            else:
+                sort_dict = {sort_prop: sort_dir}
+            block.append(sort_dict)
+        if len(block) == 0 and self.default_sort is not None:
+            block.append(self.default_sort)
+        return block
```

Comment:
Difficult to tell what changed here that is B5 related and what is code refactoring. Could those be split into separate commits or is the necessarily a complicated transformation for B5 support?

Reply:
the whole sorting block changed, I'm not sure I can split it up any further? There is no other way to have a diff

Reply:
also FYI this isn't B5 related but Datatables 1.10+ upgrade related, so there isn't going to ever be a clean-cut diff as your would expect with property changes for B5-type changes.

our old version of datatables is not compatible with bootstrap 5, which is why the upgrade was necessary. So any b5 migration of a report, will also require these kind of sorting block updates (if there is a custom sorting block) due to datatables. Relevant context is part of this PR, which we went over as a group https://github.com/dimagi/commcare-hq/pull/35648

Reply:
I guess i could have split out `_get_selected_app_sort_dict`, but that was the only refactoring related change, everything else was just...change